payjoin/src/send/v2/mod.rs

```diff
@@ -289,15 +296,26 @@ pub(crate) fn serialize_v2_body(
     Ok(format!("{}\n{}", base64, query_params).into_bytes())
 }
 
+/// Data required to validate the POST response.
+///
+/// This type is used to process a BIP77 POST response.
+/// Call [`Self::process_response`] on it to continue the BIP77 flow.
 pub struct V2PostContext {
-    /// The payjoin directory subdirectory to send the request to.
+    /// The payjoin endpoint to send the request to.
     pub(crate) endpoint: Url,
     pub(crate) psbt_ctx: PsbtContext,
     pub(crate) hpke_ctx: HpkeContext,
     pub(crate) ohttp_ctx: ohttp::ClientResponse,
 }
 
 impl V2PostContext {
+    /// Processes the response for the initial POST message from the sender client in the v2 Payjoin protocol.
+    ///
+    /// This function decapsulates the response using the provided OHTTP context. If the response status is successful,
+    /// it indicates that the Original PSBT been accepted. Otherwise, it returns an error with the status code.
+    ///
+    /// After this function is called, the sender can poll for a Proposal PSBT from the receiver
+    /// using the returned V2GetContext.
```

Comment:
I think we should clarify because there are 2 response statuses being parsed here. An OHTTP response (passed) and a decapsulated response. I suppose calling the decapsulated response "the encapsulated response" would also be correct to identify the same inner res, and perhaps even easier to understand. Odd how that works.

```suggestion
    /// Processes the response for the initial POST message from the sender
    /// client in the v2 Payjoin protocol.
    ///
    /// This function decapsulates the response using the provided OHTTP
    /// context. If the encapsulated response status is successful, it 
    /// indicates that the the Original PSBT been accepted. Otherwise, it
    /// returns an error with the encapsulated response status code.
    ///
    /// After this function is called, the sender can poll for a Proposal PSBT
    /// from the receiver using the returned [`V2GetContext`].
```

I also wrapped at 72 chars for legibility in editor.